tests/app/service/test_rest.py

```diff
@@ -703,6 +703,50 @@ def test_create_service_should_throw_duplicate_key_constraint_for_existing_email
             assert "Duplicate service name '{}'".format(service_name) in json_resp['message']['name']
 
 
+@pytest.mark.parametrize(
+    'email_from, should_error',
+    (
+            ('sams.sarnies', False),  # We are happy with plain ascii alnum/full-stops.
+            ('SAMS.SARNIES', True),  # We will reject anything with uppercase characters
+            ('sam\'s.sarnies', True),  # We reject punctuation other than a full-stop `.`
+            ('sams.Бутерброды', True),  # We reject unicode outside of the ascii charset
+            ('sams.ü', True),  # Even if it could theoretically be downcast to ascii
+            ('sams.u', False),  # Like this, which would be fine
+    )
+)
+def test_create_service_allows_only_lowercase_digits_and_fullstops_in_email_from(
+        client, service_factory, sample_user, email_from, should_error
+):
+    first_service = service_factory.get('First service', email_from='first.service')
+    service_name = 'First SERVICE'
+    data = {
+        'name': service_name,
+        'user_id': str(first_service.users[0].id),
+        'message_limit': 1000,
+        'restricted': False,
+        'active': False,
+        'email_from': email_from,
+        'created_by': str(sample_user.id)
+    }
+    auth_header = create_admin_authorization_header()
+    headers = [('Content-Type', 'application/json'), auth_header]
+    resp = client.post(
```

Comment:
We generally use the `admin_request` fixture to test requests from the admin app - it does some of the work for you (e.g. adding the headers), so if you use that fixture instead of `client` in the tests you can simplify them a bit

Reply:
Thanks, will update :)